utilities/PerformanceTesting/GetAverages.py

```diff
@@ -0,0 +1,181 @@
+    'value' to an empty list
+
+    :param cumulative: A dictionary which will contain results from all runs
+    of the performance tests for a particular version of Collector later in
+    the program. Here it will only contain empty lists.
+    """
+    for key in cumulative:
+        if 'value' in cumulative[key]:
+            cumulative[key]['value'] = []
+        else:
+            reset_cumulative(cumulative[key])
+
+
+def accumulate_metric_values_once(metrics, cumulative):
+    """
+    Loops through the dictionary metrics and appends any values it find to
+    corresponding lists in the dictionary cumulative
+
+    :param metrics: A dictionary representing the results of one run of StackRox
+    :param cumulative: A dictionary which will contain results from all runs
+    of the performance tests for a particular version of Collector
+    """
+    for key in metrics:
+        if 'value' in metrics[key]:
+            if metrics[key]['value'] is not None:
+                cumulative[key]['value'].append(metrics[key]['value'])
+        else:
+            accumulate_metric_values_once(metrics[key], cumulative[key])
+
+
+def accumulate_metric_values(metricss, cumulative):
+    """
+    Loops over a list of dictionaries with the same structure and creates a
+    dictionary where each element in the dictionary with the key value is a
+    list of the values found in the original list of dictionaries.
+
+    :param metricss: A list of dictionaries representing the results from
+    multiple runs of Collector and StackRox
+    :param cumulative: A dictionary which will contain results from all runs
+    of the performance tests for a particular version of Collector
+    """
+    for metrics in metricss:
+        accumulate_metric_values_once(metrics, cumulative)
+
+
+def calc_averages_from_cumulative(cumulative):
+    """
+    Computes the averages of the metrics
+
+    :param cumulative: A json where the values of the metrics are replaced by
+    lists representing the different values from different runs of Collector.
+    """
+    for key in cumulative:
+        if 'value' in cumulative[key]:
+            if len(cumulative[key]['value']) > 0:
+                cumulative[key]['value'] = mean(cumulative[key]['value'])
+            else:
+                cumulative[key]['value'] = None
+        else:
+            calc_averages_from_cumulative(cumulative[key])
+
+
+def calc_averages(file_prefix, num_files):
+    """
+    Computes the average results over multiple runs
+
+    :param file_prefix: The first part of the file path
+    that all files to be averaged have in common
+    :param num_files: The number of files to be averaged
+
+    :return: A dictionary with the average of the results of the different runs
+    """
+
+    file_paths = get_file_paths(file_prefix, num_files)
+
+    # metricss is a list of dictionaries, which represent the metrics obtained
+    # from one run of Collector
+    metricss = get_metricss(file_paths)
```

Comment:
This is not a typo. I had a difficult time coming up with a name for this variable. Initially I called it jsons, because it is a list of of jsons/dictionaries. I knew that was a bad name, but I couldn't come up with a better name. I didn't want to call it metrics, because it is not a simple list of metrics, it is a list of dictionaries, where each dictionary contains multiple metrics. In many cases I add a s to the end of a list to indicate that it is a list. For some two dimensional lists I add 2d to the end of the variable name. Here adding 2d was not appropriate as it is a more complicated data structure. Here I added two s to indicate that there is a nested structure of metrics. list_of_dict_of_metrics seems a bit silly. I am willing to use metrics. I am willing to take other suggestions as well.